Adds error handling for partially completed queries. Adds error handling for arguments of column names that don't match the table. Adds functionality to .select to take zero arguments or * and select all.

rcheman opened the pull request.

mjpg1 reviewed (commented):
I wonder if it makes more sense to call this method something like 'resetQuery' -- going to hold off on approving this though, since Alex and Jacob brought up some good points about whether or not we want the user to have this ability!

rcheman commented:
I've now added an error message that pops up when a user tries to chain on another insert, edit, or delete when there is already a sequal string started. 

Also, I accidentally added and deleted a file that Mia and I had put in for testing. It somehow got transferred over in my branches.

Comment feedback in github would be great too! It makes it a lot easier for me to remember and look back on later as well.

alexanderlin reviewed (commented):
do we still want keep the clear method?

rcheman commented:
Should be added to all other functions besides save, update. Look into throw New Error instead of console.error.

rcheman commented:
Both checkUnsentQuery and columnCheck now both throw Exceptions to the user to prompt them to fix their arguments/function call. 
I've added the checkUnsentQuery and columnCheck to all the methods that seem to need it. The join methods do take columns as inputs but they take columns from other tables as well and I need to do further research into how I can access that and error handle accordingly.
The select method now handles taking zero arguments or taking * to mean select all.

mjpg1 reviewed (commented):
This is SUCH awesome work, Rachel! There are just three places where I worry the checkColumns function may not work as is (all for the same reason): on line 204 in the select method, line 294 in the group method, and line 303 in the order method. Confusingly, even though those methods accept multiple columns, the parameter is still called "column" (which perhaps we should change!) So "column.toString()" will (when multiple columns are passed in and collected in an array using the rest operator) return something like "column1,column2,column3." I wonder if it would be possible to either a) tweak the checkColumns function to accept an array of potential column names OR b) in the 3 methods that accept multiple columns, iterate over the array of columns invoking the checkColumns method. Also there's always at least a 20% chance that I'm just being dumb and missing something obvious so APOLOGIES in advance if that's the case.

alexanderlin reviewed (approved):
model_test.ts is deleted, but we'll put it back, it shouldn't hold this pull request back any longer

mjpg1 commented on src/functions/errorMessages.ts:
```diff
@@ -0,0 +1,40 @@
+// checks that the user is accessing a column that already exists in the model
+export function checkColumns(columns, input: string | string[]): boolean {
+  const columnsArr: string[] = Object.keys(columns);
+  console.log(input);
```
is this console.log for the user or the developer?
